src/core/path.ts

```diff
@@ -50,7 +50,9 @@ export function safeRemoveSync(
   try {
     Deno.removeSync(file, options);
   } catch (e) {
-    if (existsSync(file)) throw e;
+    if (existsSync(file) || !(e instanceof Deno.errors.NotFound)) {
```

Comment:
I'm a little concerned about this change - why did you have to add it?

Reply:
I added it because you suggested it in the initial PR #7921  😅 
https://github.com/quarto-dev/quarto-cli/pull/7921#pullrequestreview-1782555262

At the time, I just forgot to commit  the suggestion revision in the PR before I got it merged 🤦 
https://github.com/quarto-dev/quarto-cli/pull/7921#discussion_r1427214526
So your feedback in #7921 was not included, and I added it in this PR. 

Initially I didn't added it because I wasn't aware of `Deno.errors.NotFound`, so I trusted you and it made sensed to still want to error on other type of errors. You though it was safer. 

See initial PR change for the context 
https://github.com/quarto-dev/quarto-cli/pull/7921/files

I am fine by not adding this.

Reply:
"We're all trying to find the guy who did this." Sorry!!

The reason I don't like it (anymore?) is that it changes which exceptions fall through safeRemoveSync in more places than the fix we're making.

Reply:
yeah my bad I did not commit this last time 😞 

Let's remove it then.